pkg/idstore/store_impl.go

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package idstore
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/metrics/disabled"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/idstore"
+	"github.com/hyperledger/fabric/core/ledger/util/couchdb"
+	"github.com/hyperledger/fabric/protos/common"
+	"github.com/pkg/errors"
+	"github.com/trustbloc/fabric-peer-ext/pkg/roles"
+)
+
+var logger = flogging.MustGetLogger("idstore")
+
+const (
+	systemID      = "fabric_system_"
+	inventoryName = "inventory"
+)
+
+//Store contain couchdb instance
+type Store struct {
+	db               *couchdb.CouchDatabase
+	couchMetadataRev string
+}
+
+//OpenIDStore return id store
+func OpenIDStore(path string) idstore.IDStore {
+	couchInstance, err := createCouchInstance()
+	if err != nil {
+		logger.Error("create couchdb instance failed %s", err.Error())
+		return nil
+	}
+
+	inventoryDBName := couchdb.ConstructBlockchainDBName(systemID, inventoryName)
+	if roles.IsCommitter() {
+		return newCommitterStore(couchInstance, inventoryDBName)
+	}
+	s, err := newStore(couchInstance, inventoryDBName)
+	if err != nil {
+		logger.Error(err.Error())
+		return nil
+	}
+	return s
+}
+
+func newStore(couchInstance *couchdb.CouchInstance, dbName string) (idstore.IDStore, error) {
+	db, err := couchdb.NewCouchDatabase(couchInstance, dbName)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "create new couchdb database called [%s] failed", dbName)
+	}
+
+	dbExists, err := db.ExistsWithRetry()
+	if err != nil {
+		return nil, errors.WithMessagef(err, "check couchdb [%s] exist failed %v", dbName)
```

Comment:
remove **%v** at the end of the the message

Reply:
done